spec/helpers/provider_helper_spec.rb

```diff
@@ -1,15 +1,42 @@
 require 'rails_helper'
 
 feature 'View helpers', type: :helper do
+  let(:email) { 'ab@c.com' }
+  let(:provider) { jsonapi(:provider).to_resource }
+
   describe "#add_course_link" do
-    it "returns correct google form for accrediting providers" do
-      provider = jsonapi(:provider, accredited_body?: true).to_resource
-      expect(helper.add_course_link(provider)).to eq("<a class=\"govuk-button govuk-!-margin-bottom-2\" rel=\"noopener noreferrer\" target=\"_blank\" href=\"https://forms.gle/ktbyArGW5EyiMppf9\">Add a new course</a>")
+    it "builds a link" do
+      expect(helper.add_course_link(email, provider)).to eq("<a class=\"govuk-button govuk-!-margin-bottom-2\" rel=\"noopener noreferrer\" target=\"_blank\" href=\"#{CGI::escapeHTML(helper.add_course_url(email, provider))}\">Add a new course</a>")
+    end
+  end
+
+  describe "#add_course_url" do
+    describe "for accredited bodies" do
+      let(:provider) { jsonapi(:provider, accredited_body?: true).to_resource }
+
+      it "returns correct google form" do
+        expect(helper.add_course_url(email, provider)).to start_with(Settings.google_forms.new_course_for_accredited_bodies.url)
+        expect(helper.add_course_url(email, provider)).to include(CGI::escape(email))
+        expect(helper.add_course_url(email, provider)).to include(provider.attributes[:provider_code])
+      end
     end
 
-    it "returns correct google form for non-accrediting providers" do
-      provider = jsonapi(:provider, accredited_body?: false).to_resource
-      expect(helper.add_course_link(provider)).to eq("<a class=\"govuk-button govuk-!-margin-bottom-2\" rel=\"noopener noreferrer\" target=\"_blank\" href=\"https://forms.gle/WEokN2S4qPcPAZcr5\">Add a new course</a>")
+    describe "for non-accredited bodies" do
+      let(:provider) { jsonapi(:provider, accredited_body?: false).to_resource }
+
+      it "returns correct google form" do
+        expect(helper.add_course_url(email, provider)).to start_with(Settings.google_forms.new_course_for_unaccredited_bodies.url)
+        expect(helper.add_course_url(email, provider)).to include(CGI::escape(email))
+        expect(helper.add_course_url(email, provider)).to include(provider.attributes[:provider_code])
+      end
+    end
+  end
+
+  describe "#add_location_url" do
+    it "returns a pre-populated google form URL" do
+      expect(helper.add_location_url(email, provider)).to start_with(Settings.google_forms.add_location.url)
+      expect(helper.add_location_url(email, provider)).to include(CGI::escape(email))
```

Comment:
May as well make email with that + in it, I remember chuckling

Reply:
fixed